site/book-theme/pagetoc.js

```diff
@@ -43,6 +43,31 @@ var updateDynamicHighlight = function() {
 window.addEventListener("load", updateDynamicHighlight);
 window.addEventListener("scroll", updateDynamicHighlight);
 
+/* Style the heading that matches the URL fragment (i.e. when you click a hyperlink).
+ * - Find the element with the ":target" psuedoclass applied
+ * - Measure it's rendered height, and set the '--target-height' variable to this value.
+ * - The CSS selected by ":target" will style the horizontal highlighting bar to match this height. */
+var set_target_highlight = function(event) {
+    let newurl = '';
+    if (typeof(event.newURL) === 'undefined') {
+        // probably a "load" event
+        newurl = window.location.href;
+    } else {
+        // "hashchange" event
+        newurl = event.newURL;
+    }
+    Array.prototype.forEach.call(document.getElementsByClassName("header"), function(el) {
+        if (new URL(el.href).hash == new URL(newurl).hash) {
+            document.documentElement.style.setProperty(
+                '--target-height',
+                el.getBoundingClientRect().height + 20 + "px"
```

Comment:
Where does this 20 come from?

Reply:
In combination with [margin-top](https://github.com/lowRISC/opentitan/pull/18411/files#diff-270e09b784846a990b49d7c5553cbc3097ab1272ebd9e95804db697dd0e65e9fR78), this makes the highlight bar larger than the size of the heading itself, with approximately-equal top and bottom margins.